### ISSUE TITLE
Question: TS witj `.js` references

### ISSUE THREAD

**Issue by joscha** [denoland/dnt#413]
are transformed TS files in the output expected to have `.js` references?
E.g. TS code in `npm/...` all have imports ending in `.js`, but they don't resolve anymore. Is that a JSR specialty or an issue with dnt? I can't find any information on it.

**Comment by CompeyDev** [comment]
Facing this too - is there a config or something I need to toggle?

**Comment by dsherret** [comment]
> E.g. TS code in npm/... all have imports ending in .js, but they don't resolve anymore.

Yeah for the Node output, they're supposed to have `.js` in the module specifier because in the transpiled output they'll be referencing .js files and not .ts files. When are they not resolving?

**Comment by joscha** [comment]
> is there a config or something I need to toggle?

I think this is coming from https://github.com/denoland/wasmbuild via https://github.com/denoland/dnt/blob/00b9ab70716668ab781d0d99141e52a19e15e756/mod.ts#L350 but I am not 100% sure. I tried to find the exact spot.

> When are they not resolving?

when `skipSourceOutput` is `false`, it writes the original Typescript files into the npm module, but all their imports are transformed to reference the respective `.js` file. As those files are in different folders, they don't resolve, but it doesn't make immediate sense to me anyway why the original source would be transformed.

Basically if you have:
`foo.ts`:
```
import './bla.ts';
```

then the result of `foo.ts` inside the output folder (`./npm` or similar) is: 
```
import './bla.js';
```

**Comment by joscha** [comment]
Example here: https://gist.github.com/joscha/49e26fb6a2e3b9cbae0f1f44b5879ec6

Run:
```
deno run -A build.ts 0.1.0
cat npm/src/foo.ts
```
```

**Comment by CompeyDev** [comment]
The ESM output is fine with the .js imports, but the typechecking fails when it tries to import the deno shim or similar with a .js extension.

**Comment by igorgatis** [comment]
Sounds like a duplicate of #400

**Comment by dsherret** [comment]
Sorry, I didn't understand this one earlier. I beleive it was a bug caused by a change to make `declarationMap` true by default without shipping sources. I have changed the default to be `false`, but as a temp workaround you can set `declarationMap` to false. Really sorry for the delay on this one.
